miniml/compiler/compile.scm

```diff
@@ -1753,19 +1766,16 @@
 (define (fv-list fv) (map car (vlist->list fv)))
 (define (make-nfv fv) (fold (lambda (name i nfv) (vhash-replace name i nfv)) vlist-null fv (range 0 (length fv))))
 
-(define (compile-fundef env stacksize args basebody)
-  (let* ((arity (length args))
-         (arg-names (map compile-arg-name args (range 0 arity)))
-         (body (get-fun-body args arg-names basebody))
-         (fv (fv-list (expr-fv-binding body env arg-names)))
+(define (compile-fundef env stacksize args shape body)
```

Comment:
It seems to me this function doesn't use `shape`.

Reply:
Well I hope you are wrong, because I pass `shape` as an additional parameter to `LLetfun` just for that! I will check again (but not in the next few hours), thanks.

Reply:
It seems to me the additional parameter is for `bind-var-with-shape`, isn't it?

Reply:
Yes indeed. For this reason, shapes are used in compile-recfundefs, but not in compile-fundef, so I will remove it.

The medium-term plan is to finish lowering as a proper pass, so that shapes informations are only used during lowering and never during compilation.

(The reason I'm working on lowering again is that to implement `let-exits` I need to pass an extra argument around during compilation, and I would rather first separate lowering-only arguments from compilation-only arguments to avoid bloat.)

Reply:
I removed the unused argument (in a hurry).